File: 1829-maximum-units-on-a-truck/1829-maximum-units-on-a-truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x: -x[1])
        s = 0
        for boxNumber, boxVal in boxTypes:
            if truckSize - boxNumber >= 0:
                truckSize -= boxNumber
                s += boxNumber * boxVal
            else:
                s += truckSize * boxVal
                break
        return s

File: 1829-maximum-units-on-a-truck/1829-maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, new Comparator<int[]>() {
            public int compare(int[] a, int[] b) {
                return b[1] - a[1];
            }
        });
        int t = truckSize, sum = 0;
        for (int[] boxType : boxTypes) {
            int boxNumber = boxType[0];
            if (t - boxNumber >= 0) {
                t -= boxNumber;
                sum += boxNumber * boxType[1];
            } else if (t < boxNumber) {
                sum += t * boxType[1];
                break;
            }
        }
        return sum;
    }
}